arm-none-eabi-gcc: Some platforms not building with v6.1.1-2

Arch just gave me a shiny new version of GCC to play with and I got some issues building `examples/default` now:

```
$ make -C examples/default/ buildtest
make: Entering directory '/home/martine/Repositories/RIOT-OS/RIOT/examples/default'
Building for airfy-beacon .. success
Building for arduino-due .. success
Building for arduino-mega2560 .. success
Building for avsextrem .. success
Building for cc2538dk .. success
Building for cc2650stk .. success
Building for chronos .. success
Building for ek-lm4f120xl .. success
Building for f4vi1 .. failed
Building for fox .. failed
Building for frdm-k64f .. success
Building for iotlab-m3 .. failed
Building for limifrog-v1 .. failed
Building for mbed_lpc1768 .. success
Building for msb-430 .. success
Building for msb-430h .. success
Building for msba2 .. success
Building for msbiot .. success
Building for mulle .. success
Building for native .. success
Building for nrf51dongle .. failed
Building for nrf52dk .. success
Building for nrf6310 .. failed
Building for nucleo-f072 .. failed
Building for nucleo-f091 .. failed
Building for nucleo-f103 .. success
Building for nucleo-f303 .. failed
Building for nucleo-f334 .. failed
Building for nucleo-f401 .. failed
Building for nucleo-l1 .. failed
Building for openmote-cc2538 .. success
Building for pba-d-01-kw2x .. success
Building for pca10000 .. failed
Building for pca10005 .. success
Building for pttu .. success
Building for qemu-i386 .. success
Building for remote .. success
Building for saml21-xpro .. failed
Building for samr21-xpro .. failed
Building for slwstk6220a .. success
Building for spark-core .. success
Building for stm32f0discovery .. failed
Building for stm32f3discovery .. success
Building for stm32f4discovery .. success
Building for telosb .. success
Building for udoo .. success
Building for weio .. success
Building for wsn430-v1_3b .. success
Building for wsn430-v1_4 .. success
Building for yunjia-nrf51822 .. success
Building for z1 .. success
/home/martine/Repositories/RIOT-OS/RIOT/Makefile.buildtests:38: recipe for target 'buildtest' failed
make: *** [buildtest] Error 1
make: Leaving directory '/home/martine/Repositories/RIOT-OS/RIOT/examples/default'
```

Among the boards failing are some of our most important ones. Here's a collection of a few errors I managed to identify:

```
/home/martine/Repositories/RIOT-OS/RIOT/cpu/stm32_common/periph/dac.c: In function 'dac_set':
/home/martine/Repositories/RIOT-OS/RIOT/cpu/stm32_common/periph/dac.c:76:19: error: array subscript is above array bounds [-Werror=array-bounds]
     if (dac_config[line].chan) {
         ~~~~~~~~~~^~~~~~
/home/martine/Repositories/RIOT-OS/RIOT/cpu/stm32_common/periph/dac.c: In function 'dac_poweron':
/home/martine/Repositories/RIOT-OS/RIOT/cpu/stm32_common/periph/dac.c:91:38: error: array subscript is above array bounds [-Werror=array-bounds]
     DAC->CR |= (1 << (16 * dac_config[line].chan));
                            ~~~~~~~~~~^~~~~~
/home/martine/Repositories/RIOT-OS/RIOT/cpu/stm32_common/periph/dac.c: In function 'dac_poweroff':
/home/martine/Repositories/RIOT-OS/RIOT/cpu/stm32_common/periph/dac.c:96:39: error: array subscript is above array bounds [-Werror=array-bounds]
     DAC->CR &= ~(1 << (16 * dac_config[line].chan));
                             ~~~~~~~~~~^~~~~~
cc1: all warnings being treated as errors
```

```
/home/martine/Repositories/RIOT-OS/RIOT/cpu/samd21/include/samd21.h:230:0: error: "LITTLE_ENDIAN" redefined [-Werror]
 #define LITTLE_ENDIAN          1

In file included from /usr/arm-none-eabi/include/sys/types.h:67:0,
                 from /usr/arm-none-eabi/include/stdio.h:48,
                 from /home/martine/Repositories/RIOT-OS/RIOT/examples/default/main.c:25:
/usr/arm-none-eabi/include/machine/endian.h:17:0: note: this is the location of the previous definition
 #define LITTLE_ENDIAN _LITTLE_ENDIAN

cc1: all warnings being treated as errors
```

```
/home/martine/Repositories/RIOT-OS/RIOT/cpu/nrf51/periph/adc.c: In function 'adc_sample':
/home/martine/Repositories/RIOT-OS/RIOT/cpu/nrf51/periph/adc.c:80:41: error: array subscript is above array bounds [-Werror=array-bounds]
                        (1 << (adc_config[line] + 8)) |
                               ~~~~~~~~~~^~~~~~
cc1: all warnings being treated as errors
```

Obvious workaround for now is not to update, but for the future I think it's beneficial (also with #1121 in mind) to fix those warnings.

--- a/cpu/nrf51/periph/adc.c
+++ b/cpu/nrf51/periph/adc.c
@@ -25,15 +25,11 @@
 #include "periph/adc.h"
 #include "periph_conf.h"
 
+#ifdef ADC_CONFIG
 /**
  * @brief   Load the ADC configuration
- * @{
  */
-#ifdef ADC_CONFIG
 static const uint8_t adc_config[] = ADC_CONFIG;
-#else
-static const uint8_t adc_config[] = {};
-#endif
 
 /**
  * @brief   Lock to prevent concurrency issues when used from different threads
@@ -91,3 +87,7 @@ int adc_sample(adc_t line, adc_res_t res)
 
     return val;
 }
+
+#else
+typedef int dont_be_pedantic;
+#endif /* ADC_CONFIG */

--- a/cpu/stm32_common/periph/dac.c
+++ b/cpu/stm32_common/periph/dac.c
@@ -25,7 +25,7 @@
 #include "periph_conf.h"
 
 /* only compile this, if the CPU has a DAC */
-#if defined(DAC) || defined(DAC1)
+#if (defined(DAC) || defined(DAC1)) && defined(DAC_CONFIG)
 
 #ifdef DAC2
 #define _DAC(line)          (dac_config[line].dac ? DAC2 : DAC1)
@@ -34,15 +34,9 @@
 #endif
 
 /**
- * @brief   Get the DAC configuration from the board (if configured)
- * @{
+ * @brief   Get the DAC configuration from the board config
  */
-#ifdef DAC_CONFIG
 static const dac_conf_t dac_config[] = DAC_CONFIG;
-#else
-static const dac_conf_t dac_config[] = {};
-#endif
-/** @} */
 
 int8_t dac_init(dac_t line)
 {
@@ -96,4 +90,6 @@ void dac_poweroff(dac_t line)
     DAC->CR &= ~(1 << (16 * dac_config[line].chan));
 }
 
+#else
+typedef int dont_be_pedantic;
 #endif /* DAC */

--- a/cpu/stm32f0/periph/adc.c
+++ b/cpu/stm32f0/periph/adc.c
@@ -22,6 +22,8 @@
 #include "mutex.h"
 #include "periph/adc.h"
 
+#ifdef ADC_CONFIG
+
 /**
  * @brief   Maximum allowed ADC clock speed
  */
@@ -29,13 +31,8 @@
 
 /**
  * @brief   Load the ADC configuration
- * @{
  */
-#ifdef ADC_CONFIG
 static const adc_conf_t adc_config[] = ADC_CONFIG;
-#else
-static const adc_conf_t adc_config[] = {};
-#endif
 
 /**
  * @brief   Allocate locks for all three available ADC device
@@ -105,3 +102,7 @@ int adc_sample(adc_t line,  adc_res_t res)
 
     return sample;
 }
+
+#else
+typedef int dont_be_pedantic;
+#endif /* ADC_CONFIG */

--- a/cpu/stm32f1/periph/adc.c
+++ b/cpu/stm32f1/periph/adc.c
@@ -24,6 +24,8 @@
 #include "periph/adc.h"
 #include "periph_conf.h"
 
+#ifdef ADC_CONFIG
+
 /**
  * @brief   Maximum allowed ADC clock speed
  */
@@ -31,13 +33,8 @@
 
 /**
  * @brief   Load the ADC configuration
- * @{
  */
-#ifdef ADC_CONFIG
 static const adc_conf_t adc_config[] = ADC_CONFIG;
-#else
-static const adc_conf_t adc_config[] = {};
-#endif
 
 /**
  * @brief   Allocate locks for all three available ADC devices
@@ -159,3 +156,7 @@ int adc_sample(adc_t line, adc_res_t res)
 
     return sample;
 }
+
+#else
+typedef int dont_be_pedantic;
+#endif /* ADC_CONFIG */

--- a/cpu/stm32f4/periph/adc.c
+++ b/cpu/stm32f4/periph/adc.c
@@ -23,6 +23,8 @@
 #include "periph/adc.h"
 #include "periph_conf.h"
 
+#ifdef ADC_CONFIG
+
 /**
  * @brief   Maximum allowed ADC clock speed
  */
@@ -30,13 +32,8 @@
 
 /**
  * @brief   Load the ADC configuration
- * @{
  */
-#ifdef ADC_CONFIG
 static const adc_conf_t adc_config[] = ADC_CONFIG;
-#else
-static const adc_conf_t adc_config[] = {};
-#endif
 
 /**
  * @brief   Allocate locks for all three available ADC devices
@@ -124,3 +121,7 @@ int adc_sample(adc_t line, adc_res_t res)
 
     return sample;
 }
+
+#else
+typedef int dont_be_pedantic;
+#endif /* ADC_CONFIG */

